c10/util/LeftRight.h

```diff
@@ -4,7 +4,6 @@
 #include <thread>
 
 namespace c10 {
-namespace details {
```

Comment:
I feel like there is some conversation to be had about when a file goes in `c10/util` versus `c10/detail`, and what the namespacing should be.